src/schema/core.yaml

```diff
@@ -207,6 +207,19 @@ classes:
           syntax: "{id_nmdc_prefix}:frsite-{id_shoulder}-{id_blade}$"
           interpolated: true
 
+  Sample:
+    class_uri: nmdc:Sample
+    is_a: MaterialEntity
+    description: >-
+    slot_usage:
+      id:
+        required: true
+        description: An NMDC assigned unique identifier for a biosample or processed sample.
+        structured_pattern:
+          syntax: "{id_nmdc_prefix}:(bsm|procsm)-{id_shoulder}-{id_blade}$"
```

Comment:
When the minter (part of the Runtime) sees a typecode portion that looks like this — "`(bsm|procsm)`" — it uses the first substring (`bsm`, in this case) when generating an `id` for that class. Different parts of our people/system are using the pattern for two different things: team members use it to say "here's a pattern that all IDs must match" and the minter uses it to say "here's the typecode I will use in the ID I generate". I think there is a ticket about expressing the typecode for the minter via a distinct slot. I will try to find the ticket now (I'll update this comment).